File: Two Sum II - Input array is sorted/Two_Sum_II_-_Input_array_is_sorted.py
# Runtime: 36 ms, faster than 99.51% of Python3 online submissions for Two Sum II - Input array is sorted.
# Memory Usage: 12.6 MB, less than 100.00% of Python3 online submissions for Two Sum II - Input array is sorted.

class Solution:
    def twoSum(self, numbers: 'List[int]', target: 'int') -> 'List[int]':
        res = []
        leftPtr, rightPtr = 0, len(numbers) - 1

        while leftPtr < rightPtr:
            temp = numbers[leftPtr] + numbers[rightPtr]
            if (temp == target):
                res.append(leftPtr + 1)
                res.append(rightPtr + 1)
                break
            elif temp > target:
                rightPtr -= 1
            else:
                leftPtr += 1
        
        return res

File: Two Sum II - Input array is sorted/Two_Sum_II_-_Input_array_is_sorted.cpp
// Runtime: 8 ms, faster than 100.00% of C++ online submissions for Two Sum II - Input array is sorted.
// Memory Usage: 9.6 MB, less than 100.00% of C++ online submissions for Two Sum II - Input array is sorted.

class Solution 
{
public:
    vector<int> twoSum(vector<int>& numbers, int target) 
    {
        vector<int> res;
        int leftPtr = 0, rightPtr = numbers.size() - 1;
        
        while (leftPtr < rightPtr)
        {
            int temp = numbers[leftPtr] + numbers[rightPtr];
            if (temp == target)
            {
                res.push_back(leftPtr + 1);
                res.push_back(rightPtr + 1);
                break;
            }
            else if (temp > target)
                rightPtr--;
            else 
                leftPtr++;
        }
        return res;
    }
};